controller/script/simulate-proxy/main.go

```diff
@@ -198,7 +274,47 @@ func main() {
 								HttpStatusCode: http.StatusOK,
 							},
 							ResponseLatencies: randomLatencies(count),
-							Ends:              randomEos(count),
+							Ends:              randomGrpcEos(count),
+						},
+					},
+				},
+			},
+		}
+
+		_, err = client.Report(context.Background(), req)
+		if err != nil {
+			log.Fatal(err.Error())
+		}
+
+		// HTTP/2
+		req = &pb.ReportRequest{
+			Process: &pb.Process{
+				ScheduledInstance:  "hello-h2-1mfa0",
+				ScheduledNamespace: "people-h2",
+			},
+			ClientTransports: []*pb.ClientTransport{},
+			ServerTransports: []*pb.ServerTransport{},
+			Proxy:            pb.ReportRequest_INBOUND,
+			Requests: []*pb.RequestScope{
+				&pb.RequestScope{
+					Ctx: &pb.RequestCtx{
+						SourceIp: sourceIp,
+						TargetAddr: &common.TcpAddress{
+							Ip:   targetIp,
+							Port: randomPort(),
+						},
+						Authority: "world.greeting:7778",
+						Method:    &common.HttpMethod{Type: &common.HttpMethod_Registered_{Registered: common.HttpMethod_GET}},
+						Path:      "/World/GreetingH2",
+					},
+					Count: count,
+					Responses: []*pb.ResponseScope{
+						&pb.ResponseScope{
+							Ctx: &pb.ResponseCtx{
+								HttpStatusCode: randomHttpResponseCode(),
+							},
+							ResponseLatencies: randomLatencies(count),
+							Ends:              randomH2Eos(count),
 						},
 					},
 				},
```

Comment:
There's a lot of duplication in the two `ReportRequest` structs now defined in this file. I think it would be better to call the `Report` endpoint only once, with a `ReportRequest` struct that includes both a gRPC request report and an H2 request report. Specifically, The `ReportRequest.Requests` field sends a slice of `RequestScope` objects, so rather that sending two separate reports, each with one `RequestScope` object, I'd send one report with two `RequestScope` objects.

Reply:
@klingerf I think 3001ccf82e5049e42d1e8c1bff4ffaa3dae15d8a makes the change you've requested here? I also moved the TCP telemetry added in #165, I hope that's not out of scope for this PR.